projects/frontend/data-pipelines/gui/projects/ui/src/app/app-config.service.ts

```diff
@@ -8,22 +8,28 @@ import { HttpBackend, HttpClient } from '@angular/common/http';
 import { AuthConfig } from 'angular-oauth2-oidc';
 import { AppConfig, RefreshTokenConfig } from './app-config.model';
 import { firstValueFrom } from 'rxjs';
-
+import { Router } from '@angular/router';
+import { routes } from './app.routing';
 @Injectable({
     providedIn: 'root'
 })
 export class AppConfigService {
     private httpClient: HttpClient;
     private appConfig: AppConfig;
+    private router: Router;
 
-    constructor(private httpBackend: HttpBackend) {
+    constructor(private httpBackend: HttpBackend, private angularRouter: Router) {
```

Comment:
Why declaring one class field `angularRouter` of type `Router` that is after that assigned to class field `router` of the same type `Router`, and then using the second one?
Adding modifier of type `public|protected|private` in class constructor in Typescript will create class field with the same name that later could be use in the class lifecycle for any type of Typescript class (Component|Service|Directive|pure Typescript/JavaScript class|etc...)

Reply:
I'll remove the class field. I did this in the interest of testability. I was hoping to test the dynamic routes using an angular spec. I ended up moving it to the cypress tests, so no need for making the router testable.